### PR TITLE
Sound: Add pitch option

### PR DESCRIPTION
**Pull request by Rui-Minetest**
```
minetest.sound_play("sound", {
    pitch = 1.5
})

minetest.register_node("aaa:aaa", {
    sounds = {
        footstep = {name = "default_place_node", gain = 1.0, pitch = 1.5}
    }
})
```

### CODE REVIEW

**Comment by bigfoot547** [comment]
This would make noteblocks a lot easier

File: src/network/clientpackethandler.cpp [hunk 1]
```diff
@@ -775,28 +776,30 @@ void Client::handleCommand_PlaySound(NetworkPacket* pkt)
 	u16 object_id;
 	bool loop;
 	float fade = 0;
+	float pitch = 1.0;
```
Comment (nerzhul) [inline src/network/clientpackethandler.cpp]: 1.0f

File: src/script/lua_api/l_client.cpp [hunk 2]
```diff
@@ -228,25 +228,27 @@ int ModApiClient::l_sound_play(lua_State *L)
 	SimpleSoundSpec spec;
 	read_soundspec(L, 1, spec);
 	float gain = 1.0;
+	float pitch = 1.0;
```
Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: same here

File: src/server.h [hunk 3]
```diff
@@ -116,6 +116,7 @@ struct ServerSoundParams
 	float max_hear_distance;
 	bool loop;
 	float fade;
+	float pitch;
```
Comment (nerzhul) [inline src/server.h]: now we are using C++11 you can do "float pitch = 1.0f;"  directly, maybe convert the whole constructor in that style ? :)

File: src/sound.h [hunk 4]
```diff
@@ -44,6 +45,7 @@ struct SimpleSoundSpec
 	std::string name;
 	float gain;
 	float fade;
+	float pitch;
```
Comment (nerzhul) [inline src/sound.h]: same for those params, please set default values here (you cannot remove the params affectations)

**Comment by i17305** [comment]
Updated.

**Comment by Rui-Minetest** [comment]
Updated.

**Comment by nerzhul** [comment]
can you fix lint ? (look at travis)

**Comment by Fixer-007** [comment]
Introduced regression: can't connect to other servers now (0.4.14-16), singleplayer works:
```
ERROR[Main]: A serialization error occurred:
ERROR[Main]: deSerializeString: couldn't read all chars
ERROR[Main]: The server is probably  running a different version of Minetest.
```

**Comment by Rui-Minetest** [comment]
#5964

**Comment by Fixer-007** [comment]
Confirm it is fixed by 5964

**Pull request merged**